wavecrestweb/kenworthy-machine
# #KM-18/feature card layout

## Description

This PR adds the Card component with general layout for the front of the card. I imported Card component right after CompanyValues component while I work on styling it (I'll remove it once I'm done with the component).

## WIP
Styling is still work in progress and I will abstract logic (including styling) for corner label in a separate component to clean up the card component.
Responsiveness for small/medium screen sizes will be implemented next.

## Related Tickets & Documents
Closes #KM-18

## Acceptance Criteria

- [x] The card has a dedicated section for the machine photo.
- [x] The card has a machine label.
- [x] The card has a machine title.
- [x] The card has a button to flip it.
- [x] The card is designed for easy readability, with font sizes and colors that match the overall aesthetic of the website.

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|  `✓`   | :sparkles: New feature     |
|    `✓` | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings
![Screen Shot 2024-02-04 at 9 38 28 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/f8bf0140-5b78-4f00-94fe-1cab21779ae4)


## What I learned
I learned more about customizing theme for multipart-components (specifically for Card component) using cardAnatomy and createMultiStyleConfigHelpers. I also discovered that everything that has to do with sizing (borderRadius, padding, margins, width, height, etc) can be overwritten only within sizes object for the card component (it's also true for some other components from what I read). 
Also, had to look up how to create triangular shape for the label with css. Still trying to decide if creating corner label with css only is better than figuring out how to overlap png file with label over the image and adding text on top of it all.

## Testing steps
1. Pull down and check out this branch
2. Run `npm run dev`
3. Open http://localhost:3000/ in your browser of choice
4. Scroll past Company Values section to see the card

## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
|  `✓`   | 🙅 no documentation needed    |


## What gif best describes this PR?
